// @muhlba91/pulumi-proxmoxve/network/firewallOptions.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirewallOptions = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages firewall options on VM / Container level.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as proxmoxve from "@muhlba91/pulumi-proxmoxve";
 *
 * const example = new proxmoxve.network.FirewallOptions("example", {
 *     nodeName: proxmox_virtual_environment_vm.example.node_name,
 *     vmId: proxmox_virtual_environment_vm.example.vm_id,
 *     dhcp: true,
 *     enabled: false,
 *     ipfilter: true,
 *     logLevelIn: "info",
 *     logLevelOut: "info",
 *     macfilter: false,
 *     ndp: true,
 *     inputPolicy: "ACCEPT",
 *     outputPolicy: "ACCEPT",
 *     radv: true,
 * }, {
 *     dependsOn: [proxmox_virtual_environment_vm.example],
 * });
 * ```
 */
class FirewallOptions extends pulumi.CustomResource {
    /**
     * Get an existing FirewallOptions resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new FirewallOptions(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FirewallOptions.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FirewallOptions.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["containerId"] = state ? state.containerId : undefined;
            resourceInputs["dhcp"] = state ? state.dhcp : undefined;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["inputPolicy"] = state ? state.inputPolicy : undefined;
            resourceInputs["ipfilter"] = state ? state.ipfilter : undefined;
            resourceInputs["logLevelIn"] = state ? state.logLevelIn : undefined;
            resourceInputs["logLevelOut"] = state ? state.logLevelOut : undefined;
            resourceInputs["macfilter"] = state ? state.macfilter : undefined;
            resourceInputs["ndp"] = state ? state.ndp : undefined;
            resourceInputs["nodeName"] = state ? state.nodeName : undefined;
            resourceInputs["outputPolicy"] = state ? state.outputPolicy : undefined;
            resourceInputs["radv"] = state ? state.radv : undefined;
            resourceInputs["vmId"] = state ? state.vmId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.nodeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'nodeName'");
            }
            resourceInputs["containerId"] = args ? args.containerId : undefined;
            resourceInputs["dhcp"] = args ? args.dhcp : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["inputPolicy"] = args ? args.inputPolicy : undefined;
            resourceInputs["ipfilter"] = args ? args.ipfilter : undefined;
            resourceInputs["logLevelIn"] = args ? args.logLevelIn : undefined;
            resourceInputs["logLevelOut"] = args ? args.logLevelOut : undefined;
            resourceInputs["macfilter"] = args ? args.macfilter : undefined;
            resourceInputs["ndp"] = args ? args.ndp : undefined;
            resourceInputs["nodeName"] = args ? args.nodeName : undefined;
            resourceInputs["outputPolicy"] = args ? args.outputPolicy : undefined;
            resourceInputs["radv"] = args ? args.radv : undefined;
            resourceInputs["vmId"] = args ? args.vmId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(FirewallOptions.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FirewallOptions = FirewallOptions;
/** @internal */
FirewallOptions.__pulumiType = 'proxmoxve:Network/firewallOptions:FirewallOptions';
