// @muhlba91/pulumi-proxmoxve/apt/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.standard = exports.Repository = exports.getRepositoryOutput = exports.getRepository = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getRepository = null;
exports.getRepositoryOutput = null;
utilities.lazyLoad(exports, ["getRepository", "getRepositoryOutput"], () => require("./getRepository"));
exports.Repository = null;
utilities.lazyLoad(exports, ["Repository"], () => require("./repository"));
// Export sub-modules:
const standard = require("./standard");
exports.standard = standard;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "proxmoxve:Apt/repository:Repository":
                return new exports.Repository(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("proxmoxve", "Apt/repository", _module);
