// @muhlba91/pulumi-proxmoxve/config/vars.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const __config = new pulumi.Config("proxmoxve");
Object.defineProperty(exports, "apiToken", {
    get() {
        return __config.get("apiToken");
    },
    enumerable: true,
});
Object.defineProperty(exports, "endpoint", {
    get() {
        return __config.get("endpoint");
    },
    enumerable: true,
});
Object.defineProperty(exports, "insecure", {
    get() {
        return __config.getObject("insecure");
    },
    enumerable: true,
});
Object.defineProperty(exports, "minTls", {
    get() {
        return __config.get("minTls");
    },
    enumerable: true,
});
Object.defineProperty(exports, "otp", {
    get() {
        return __config.get("otp");
    },
    enumerable: true,
});
Object.defineProperty(exports, "password", {
    get() {
        return __config.get("password");
    },
    enumerable: true,
});
Object.defineProperty(exports, "ssh", {
    get() {
        return __config.getObject("ssh");
    },
    enumerable: true,
});
Object.defineProperty(exports, "tmpDir", {
    get() {
        return __config.get("tmpDir");
    },
    enumerable: true,
});
Object.defineProperty(exports, "username", {
    get() {
        return __config.get("username");
    },
    enumerable: true,
});
