// @muhlba91/pulumi-proxmoxve/acl.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Acl = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Manages ACLs on the Proxmox cluster.
 *
 * ACLs are used to control access to resources in the Proxmox cluster.
 * Each ACL consists of a path, a user, group or token, a role, and a flag to allow propagation of permissions.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as proxmoxve from "@muhlba91/pulumi-proxmoxve";
 *
 * const operationsAutomation = new proxmoxve.permission.User("operationsAutomation", {
 *     comment: "Managed by Terraform",
 *     password: "a-strong-password",
 *     userId: "operations-automation@pve",
 * });
 * const operationsMonitoring = new proxmoxve.permission.Role("operationsMonitoring", {
 *     roleId: "operations-monitoring",
 *     privileges: ["VM.Monitor"],
 * });
 * const operationsAutomationMonitoring = new proxmoxve.Acl("operationsAutomationMonitoring", {
 *     userId: operationsAutomation.userId,
 *     roleId: operationsMonitoring.roleId,
 *     path: "/vms/1234",
 *     propagate: true,
 * });
 * ```
 *
 * ## Import
 *
 * #!/usr/bin/env sh
 *
 * ACL can be imported using its unique identifier, e.g.: {path}?{group|user@realm|user@realm!token}?{role}
 *
 * ```sh
 * $ pulumi import proxmoxve:index/acl:Acl operations_automation_monitoring /?monitor@pve?operations-monitoring
 * ```
 */
class Acl extends pulumi.CustomResource {
    /**
     * Get an existing Acl resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Acl(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Acl.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Acl.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["groupId"] = state ? state.groupId : undefined;
            resourceInputs["path"] = state ? state.path : undefined;
            resourceInputs["propagate"] = state ? state.propagate : undefined;
            resourceInputs["roleId"] = state ? state.roleId : undefined;
            resourceInputs["tokenId"] = state ? state.tokenId : undefined;
            resourceInputs["userId"] = state ? state.userId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.path === undefined) && !opts.urn) {
                throw new Error("Missing required property 'path'");
            }
            if ((!args || args.roleId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleId'");
            }
            resourceInputs["groupId"] = args ? args.groupId : undefined;
            resourceInputs["path"] = args ? args.path : undefined;
            resourceInputs["propagate"] = args ? args.propagate : undefined;
            resourceInputs["roleId"] = args ? args.roleId : undefined;
            resourceInputs["tokenId"] = args ? args.tokenId : undefined;
            resourceInputs["userId"] = args ? args.userId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Acl.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Acl = Acl;
/** @internal */
Acl.__pulumiType = 'proxmoxve:index/acl:Acl';
