// @muhlba91/pulumi-proxmoxve/hosts.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Hosts = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Manages the host entries on a specific node.
 *
 * ## Import
 *
 * Instances can be imported using the `node_name`, e.g.,
 *
 * bash
 *
 * ```sh
 * $ pulumi import proxmoxve:index/hosts:Hosts first_node_host_entries first-node
 * ```
 */
class Hosts extends pulumi.CustomResource {
    /**
     * Get an existing Hosts resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Hosts(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Hosts.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Hosts.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["addresses"] = state ? state.addresses : undefined;
            resourceInputs["digest"] = state ? state.digest : undefined;
            resourceInputs["entries"] = state ? state.entries : undefined;
            resourceInputs["entry"] = state ? state.entry : undefined;
            resourceInputs["hostnames"] = state ? state.hostnames : undefined;
            resourceInputs["nodeName"] = state ? state.nodeName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.entry === undefined) && !opts.urn) {
                throw new Error("Missing required property 'entry'");
            }
            if ((!args || args.nodeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'nodeName'");
            }
            resourceInputs["entry"] = args ? args.entry : undefined;
            resourceInputs["nodeName"] = args ? args.nodeName : undefined;
            resourceInputs["addresses"] = undefined /*out*/;
            resourceInputs["digest"] = undefined /*out*/;
            resourceInputs["entries"] = undefined /*out*/;
            resourceInputs["hostnames"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Hosts.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Hosts = Hosts;
/** @internal */
Hosts.__pulumiType = 'proxmoxve:index/hosts:Hosts';
