// @muhlba91/pulumi-proxmoxve/time.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Time = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Manages the time for a specific node.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as proxmoxve from "@muhlba91/pulumi-proxmoxve";
 *
 * const firstNodeTime = new proxmoxve.Time("firstNodeTime", {
 *     nodeName: "first-node",
 *     timeZone: "UTC",
 * });
 * ```
 *
 * ## Import
 *
 * Instances can be imported using the `node_name`, e.g.,
 *
 * bash
 *
 * ```sh
 * $ pulumi import proxmoxve:index/time:Time first_node first-node
 * ```
 */
class Time extends pulumi.CustomResource {
    /**
     * Get an existing Time resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Time(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Time.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Time.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["localTime"] = state ? state.localTime : undefined;
            resourceInputs["nodeName"] = state ? state.nodeName : undefined;
            resourceInputs["timeZone"] = state ? state.timeZone : undefined;
            resourceInputs["utcTime"] = state ? state.utcTime : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.nodeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'nodeName'");
            }
            if ((!args || args.timeZone === undefined) && !opts.urn) {
                throw new Error("Missing required property 'timeZone'");
            }
            resourceInputs["nodeName"] = args ? args.nodeName : undefined;
            resourceInputs["timeZone"] = args ? args.timeZone : undefined;
            resourceInputs["localTime"] = undefined /*out*/;
            resourceInputs["utcTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Time.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Time = Time;
/** @internal */
Time.__pulumiType = 'proxmoxve:index/time:Time';
