// @muhlba91/pulumi-proxmoxve/hardware/mapping/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Usb = exports.Pci = exports.getUsbOutput = exports.getUsb = exports.getPciOutput = exports.getPci = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getPci = null;
exports.getPciOutput = null;
utilities.lazyLoad(exports, ["getPci", "getPciOutput"], () => require("./getPci"));
exports.getUsb = null;
exports.getUsbOutput = null;
utilities.lazyLoad(exports, ["getUsb", "getUsbOutput"], () => require("./getUsb"));
exports.Pci = null;
utilities.lazyLoad(exports, ["Pci"], () => require("./pci"));
exports.Usb = null;
utilities.lazyLoad(exports, ["Usb"], () => require("./usb"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "proxmoxve:Hardware/mapping/pci:Pci":
                return new exports.Pci(name, undefined, { urn });
            case "proxmoxve:Hardware/mapping/usb:Usb":
                return new exports.Usb(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("proxmoxve", "Hardware/mapping/pci", _module);
pulumi.runtime.registerResourceModule("proxmoxve", "Hardware/mapping/usb", _module);
