// @muhlba91/pulumi-proxmoxve/vm/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachine2 = exports.VirtualMachine = exports.getVirtualMachinesOutput = exports.getVirtualMachines = exports.getVirtualMachineOutput = exports.getVirtualMachine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getVirtualMachine = null;
exports.getVirtualMachineOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachine", "getVirtualMachineOutput"], () => require("./getVirtualMachine"));
exports.getVirtualMachines = null;
exports.getVirtualMachinesOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachines", "getVirtualMachinesOutput"], () => require("./getVirtualMachines"));
exports.VirtualMachine = null;
utilities.lazyLoad(exports, ["VirtualMachine"], () => require("./virtualMachine"));
exports.VirtualMachine2 = null;
utilities.lazyLoad(exports, ["VirtualMachine2"], () => require("./virtualMachine2"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "proxmoxve:VM/virtualMachine2:VirtualMachine2":
                return new exports.VirtualMachine2(name, undefined, { urn });
            case "proxmoxve:VM/virtualMachine:VirtualMachine":
                return new exports.VirtualMachine(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("proxmoxve", "VM/virtualMachine", _module);
pulumi.runtime.registerResourceModule("proxmoxve", "VM/virtualMachine2", _module);
