// @muhlba91/pulumi-proxmoxve/ha/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HAResource = exports.HAGroup = exports.getHAResourcesOutput = exports.getHAResources = exports.getHAResourceOutput = exports.getHAResource = exports.getHAGroupsOutput = exports.getHAGroups = exports.getHAGroupOutput = exports.getHAGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getHAGroup = null;
exports.getHAGroupOutput = null;
utilities.lazyLoad(exports, ["getHAGroup", "getHAGroupOutput"], () => require("./getHAGroup"));
exports.getHAGroups = null;
exports.getHAGroupsOutput = null;
utilities.lazyLoad(exports, ["getHAGroups", "getHAGroupsOutput"], () => require("./getHAGroups"));
exports.getHAResource = null;
exports.getHAResourceOutput = null;
utilities.lazyLoad(exports, ["getHAResource", "getHAResourceOutput"], () => require("./getHAResource"));
exports.getHAResources = null;
exports.getHAResourcesOutput = null;
utilities.lazyLoad(exports, ["getHAResources", "getHAResourcesOutput"], () => require("./getHAResources"));
exports.HAGroup = null;
utilities.lazyLoad(exports, ["HAGroup"], () => require("./hagroup"));
exports.HAResource = null;
utilities.lazyLoad(exports, ["HAResource"], () => require("./haresource"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "proxmoxve:HA/hAGroup:HAGroup":
                return new exports.HAGroup(name, undefined, { urn });
            case "proxmoxve:HA/hAResource:HAResource":
                return new exports.HAResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("proxmoxve", "HA/hAGroup", _module);
pulumi.runtime.registerResourceModule("proxmoxve", "HA/hAResource", _module);
