// @muhlba91/pulumi-proxmoxve/permission/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.Role = exports.Pool = exports.Group = exports.getUsersOutput = exports.getUsers = exports.getUserOutput = exports.getUser = exports.getRolesOutput = exports.getRoles = exports.getRoleOutput = exports.getRole = exports.getPoolsOutput = exports.getPools = exports.getPoolOutput = exports.getPool = exports.getGroupsOutput = exports.getGroups = exports.getGroupOutput = exports.getGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGroup = null;
exports.getGroupOutput = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
exports.getGroups = null;
exports.getGroupsOutput = null;
utilities.lazyLoad(exports, ["getGroups", "getGroupsOutput"], () => require("./getGroups"));
exports.getPool = null;
exports.getPoolOutput = null;
utilities.lazyLoad(exports, ["getPool", "getPoolOutput"], () => require("./getPool"));
exports.getPools = null;
exports.getPoolsOutput = null;
utilities.lazyLoad(exports, ["getPools", "getPoolsOutput"], () => require("./getPools"));
exports.getRole = null;
exports.getRoleOutput = null;
utilities.lazyLoad(exports, ["getRole", "getRoleOutput"], () => require("./getRole"));
exports.getRoles = null;
exports.getRolesOutput = null;
utilities.lazyLoad(exports, ["getRoles", "getRolesOutput"], () => require("./getRoles"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.getUsers = null;
exports.getUsersOutput = null;
utilities.lazyLoad(exports, ["getUsers", "getUsersOutput"], () => require("./getUsers"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.Pool = null;
utilities.lazyLoad(exports, ["Pool"], () => require("./pool"));
exports.Role = null;
utilities.lazyLoad(exports, ["Role"], () => require("./role"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "proxmoxve:Permission/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "proxmoxve:Permission/pool:Pool":
                return new exports.Pool(name, undefined, { urn });
            case "proxmoxve:Permission/role:Role":
                return new exports.Role(name, undefined, { urn });
            case "proxmoxve:Permission/user:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("proxmoxve", "Permission/group", _module);
pulumi.runtime.registerResourceModule("proxmoxve", "Permission/pool", _module);
pulumi.runtime.registerResourceModule("proxmoxve", "Permission/role", _module);
pulumi.runtime.registerResourceModule("proxmoxve", "Permission/user", _module);
